src/main/java/sg/dex/starfish/keeper/DirectPurchaseAdapter.java

```diff
@@ -0,0 +1,111 @@
+package sg.dex.starfish.keeper;
+
+import sg.dex.starfish.impl.squid.SquidService;
+import com.oceanprotocol.common.web3.KeeperService;
+
+import com.oceanprotocol.squid.exceptions.TokenApproveException;
+import java.io.IOException;
+import org.web3j.crypto.CipherException;
+
+import org.web3j.utils.Numeric;
+import java.util.ArrayList;
+import java.math.BigInteger;
+import java.util.Properties;
+import java.io.InputStream;
+import com.oceanprotocol.keeper.contracts.OceanToken;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+
+import org.web3j.protocol.core.methods.request.EthFilter;
+import io.reactivex.Flowable;
+import org.web3j.abi.EventEncoder;
+import org.web3j.protocol.core.DefaultBlockParameter;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+
+public final class DirectPurchaseAdapter {
+    private DirectPurchase directPurchase;
+    private TokenManager tokenManager;
+
+    public DirectPurchaseAdapter() throws IOException, CipherException {
```

Comment:
Style: it is best if a constructor takes it's fields as parameters.

Any lookup of parameter values should be done outside the constructor in a static function such as `DirectPurchaseAdapter.create(....)`.

Reply:
fixed